src/engine/TransitivePathBinSearch.cpp

```diff
@@ -0,0 +1,245 @@
+      insertIntoMap(hull, startNode, startNode);
+    }
+
+    while (stack.size() > 0) {
+      checkCancellation();
+      auto [node, steps] = stack.back();
+      stack.pop_back();
+
+      if (steps <= maxDist_ && marks.count(node) == 0) {
+        if (steps >= minDist_) {
+          marks.insert(node);
+          if (!target.has_value() || node == target.value()) {
+            insertIntoMap(hull, startNode, node);
+          }
+        }
+
+        auto successors = edges.successors(node);
+        for (auto successor : successors) {
+          stack.push_back({successor, steps + 1});
+        }
+      }
+    }
+  }
+  return hull;
+}
+
+// _____________________________________________________________________________
+template <size_t SUB_WIDTH, size_t SIDE_WIDTH>
+std::pair<BinSearchMap, std::vector<Id>>
+TransitivePathBinSearch::setupMapAndNodes(const IdTable& sub,
+                                          const TransitivePathSide& startSide,
+                                          const TransitivePathSide& targetSide,
+                                          const IdTable& startSideTable) const {
+  std::vector<Id> nodes;
+  auto edges = setupEdgesMap<SUB_WIDTH>(sub, startSide, targetSide);
+
+  // Bound -> var|id
+  std::span<const Id> startNodes =
+      startSideTable.getColumn(startSide.treeAndCol_.value().second);
+  nodes.insert(nodes.end(), startNodes.begin(), startNodes.end());
+
+  return {std::move(edges), std::move(nodes)};
+}
+
+// _____________________________________________________________________________
+template <size_t SUB_WIDTH>
+std::pair<BinSearchMap, std::vector<Id>>
+TransitivePathBinSearch::setupMapAndNodes(
+    const IdTable& sub, const TransitivePathSide& startSide,
+    const TransitivePathSide& targetSide) const {
+  std::vector<Id> nodes;
+  auto edges = setupEdgesMap<SUB_WIDTH>(sub, startSide, targetSide);
+
+  // id -> var|id
+  if (!startSide.isVariable()) {
+    nodes.push_back(std::get<Id>(startSide.value_));
+    // var -> var
+  } else {
+    std::span<const Id> startNodes = sub.getColumn(startSide.subCol_);
+    nodes.insert(nodes.end(), startNodes.begin(), startNodes.end());
+    if (minDist_ == 0) {
+      std::span<const Id> targetNodes = sub.getColumn(targetSide.subCol_);
+      nodes.insert(nodes.end(), targetNodes.begin(), targetNodes.end());
+    }
+  }
+
+  return {std::move(edges), std::move(nodes)};
+}
+
+// _____________________________________________________________________________
+template <size_t SUB_WIDTH>
+BinSearchMap TransitivePathBinSearch::setupEdgesMap(
+    const IdTable& dynSub, const TransitivePathSide& startSide,
+    const TransitivePathSide& targetSide) const {
+  const IdTableView<SUB_WIDTH> sub = dynSub.asStaticView<SUB_WIDTH>();
+  decltype(auto) startCol = sub.getColumn(startSide.subCol_);
+  decltype(auto) targetCol = sub.getColumn(targetSide.subCol_);
+  BinSearchMap edges{startCol, targetCol};
+
+  return edges;
+}
```

Comment:
I think there is absolutely no reason for this to be a template, the `getColumn` also works without loss of performance on `IdTable`s.
Then this is a one-liner:
`return BinSearchMap{sub.getColumn(startSide.subCol), sub.getColumn(targetSide.subCol_)};